### PR TITLE
Buttons UI

### PR DESCRIPTION
**Pull request by debscorreia**
**Issue:** https://labcodes.atlassian.net/browse/DSYS-6
**Figma:** https://www.figma.com/file/Wx7jj5klQn3YsRA2XcIV3E/Design-System-MASTER?node-id=18%3A188

This PR adds the button component, its styles and the storybook story. It focus sofar on the button UI.
It also creates styles and the storybook story for Links

**Props**
Button has 3 variants: `default`, `outline` and `text` and 3 sizes: `small`, `normal` and `large`.
Buttons are made of an option button and a mandatory text label.
It can also have different skins for all variants: `light` and `dark`, and the default variant has 5 more skins: `warning`, `destructive`, `warning-invert`, `destructive-invert` and `confirmation-invert`.

Screenshots
image

image

### CODE REVIEW

File: labsystem/scss/main.scss [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 @import "variables/all";
 
-// Base elements
-@import "base/all";
-
 // Helpers
 @import "utilities/all";
 
+// Base elements
+@import "base/all";
```
Comment (lucianoratamero) [inline labsystem/scss/main.scss:8]: nice catch!

File: stories/Buttons.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+
+<h4>Labels</h4>
+<p>Button labels must say what will happen when the user interacts with it. Labels use sentence case (only the first word in a phrase and any proper nouns capitalized) and are as short as possible (max. of 3 words). They must clearly explain what happens when the button is clicked.</p>
+
+<h4>Icon use</h4>
+<ul>
+  <li>Use icon frames of 16px within buttons.</li>
+  <li>Icons should always appear to the left of the label.</li>
+  <li>Icons used in buttons must be directly related to the action that the user is taking. <span class="u-color-cyan60">Don't hesitate to create a new icon when it is necessary.</span></li>
+  <li>Always use the same color value as the label within a button. Icon follows label with equal emphasys.</li>
```
Comment (CavalcanteLucas) [inline stories/Buttons.stories.mdx:39]: I can't understand the first phrase.

File: stories/Buttons.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+  <li>Use icon in buttons to give more significancy to any action, function or indicate triggers as "open external link".</li>
+</ul>
+<p><span class="u-color-pink60">You can use an only icon button on extreme situations. Remember that, in these cases, the label is represented by the pictogram itself, so, its drawing must be clear and communicate in one glance what action it triggers.</span></p>
```
Comment (CavalcanteLucas) [inline stories/Buttons.stories.mdx:42]: suggestion:

`<p><span class="u-color-pink60">An only icon button is represented by a pictogram. Use it on extreme situations. Make sure the drawing is clear and communicate in one glance what action it triggers.</span></p>
`

File: stories/Buttons.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+          <div className="doc-buttons"><Button text="Outline" variant="outline" icon="reload" /></div>
+        </td>
+        <td><p><strong>Outline:</strong> Secondary actions or triggers on each page. Prefer this one to the default one, specially on a page with lots of buttons.</p></td>
+      </tr>
+      <tr>
+        <td>
+          <div className="doc-buttons"><Button text="Text" variant="text" /></div>
+          <div className="doc-buttons"><Button text="Text" variant="text" icon="reload" /></div>
+        </td>
+        <td><p><strong>Text:</strong> Tertiary or less important actions or triggers. Use it to connect internal pages or trigger internal functions of a component.</p></td>
```
Comment (CavalcanteLucas) [inline stories/Buttons.stories.mdx:72]: suggestion:

`        <td><p><strong>Text:</strong> Tertiary or less important actions or triggers. Use it to connect internal pages or to trigger internal functions of a component.</p></td>
`

File: stories/Buttons.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,447 @@
...
+      <li>Base border</li>
+      <li>Focus border</li>
+      <li>Hover border</li>
+      <li>Hyperlink indicator</li>
+    </ol>
+  </div>
+  <div class="column">
+    <h5>Borders</h5><br/><br/>
+    <img src="docs/buttons/btn-borders.svg" alt="Button borders scheme"/><br/><br/>
+    <p>All the buttons have a rounded corner radius of 4px on it's base background or border. Base and Focus borders should have 2px and Hover border, 6px.</p>
```
Comment (CavalcanteLucas) [inline stories/Buttons.stories.mdx:136]: suggestion:

`    <p>All the buttons have a rounded corner radius of 4px on it's base background or border. Base and Focus borders should have 2px, and Hover borders should have 6px.</p>
`

File: stories/Buttons.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,447 @@
...
+      </div>
+    </Fragment>
+  </Story>
+</Preview>
+
+#### Skins
+
+Use the mineral or white skins for primary actions on <b>toasts</b> and <b>banners</b>.
+
+You can also use <b>Message Colors</b> on labels to place these buttons also in background colored <b>notifications</b>.
```
Comment (CavalcanteLucas) [inline stories/Buttons.stories.mdx:205]: I can't understand it.

**Pull request merged**